### ISSUE TITLE
Support Pagination with Params

### ISSUE THREAD

**Issue by jpulec** [alecraso/grnhse-api#3]
It doesn't appear that this library supports passing params when trying to use the built in pagination. The `_get()` call for `get_next()` explicitly sets `params=False` which means that if you tried to filter and endpoint using a param, such as `updated_after`, it will not work correctly.

**Comment by io-ma** [comment]
So I thought I could use pagination on all endpoints and actually I am calling candidates. It works, it just gets me the whole soup and I can't pass params to it. I try passing approval_type="open_job".
I modified params=False in that get call, in both places. Not working.
Then I stumbled on this: https://developers.greenhouse.io/harvest.html#pagination.
I mean I can obviously call next pages. I'd just rather get 2000 candidates using params,  instead of 10000 :-D

**Comment by io-ma** [comment]
Also, I'd like to help if you point me in the right direction

**Comment by io-ma** [comment]
@jpulec any ideas on this one?

**Comment by jpulec** [comment]
@io-ma Not quite sure I follow what you're describing. Pagination should work by iterating over a particular endpoint, i.e. `for candidate_page in harvest.candidates`, as described in the docs. However, you can't pass any params to an endpoint and expect pagination to work. That's the issue I've described above.

As a workaround, you can just loop until the `records_remaining` attribute is no longer true:

```python
candidate_endpoint = havest.candidates()
candidates = candidate_endpoint.get()
while candidate_endpoint.records_remaining:
    candidates.extend(candidates_endpoint._get(candidates_endpoint._next_url))
```

That should allow fetching all paginated records, and allow passing any relevant params. But `approval_type` isn't a valid param for `candidates` if you check the docs.

**Comment by io-ma** [comment]
@jpulec I know, I do that and can get all the thousands of candidates.  Also, I got familiar with the params now, so indeed approval_type is not a valid one. At that time I thought the params in the Greenhouse UI were available for the API itself - you know those filters in the frontend.

![Desktop-screenshot - 2021-02-08T101507 486](https://user-images.githubusercontent.com/2306628/107192708-a40b3000-69f6-11eb-8ec2-b9582a847c64.png)

So let's recap, I do the loop, I get all candidates. But say , in the example above , when you say:

> That should allow fetching all paginated records, and allow passing any relevant params.

Where do you pass the params? Inside the `candidate_endpoint.get()`?

**Comment by jpulec** [comment]
Yep. Something like `candidate_endpoint.get(job_id=12345)`. Any keyword arguments to `.get()` are passed as url params.

**Comment by io-ma** [comment]
Ok, thank you for the info, it helps a ton! What I find annoying with Greenhouse is you can't chain the params ...

**Comment by rogusdev** [comment]
+1 on this issue. There is a workaround (as above), but that prevents us from using list comprehensions. The fix seems so easy -- just move the `params` as an extra arg with default value into the function. I did change that default to `True` for consistency with the normal `_get` behavior, and because that is the issue here:

```
def get_next(self, params=True):
        if self._next_url is None:
            raise AttributeError('Next url is not set, try querying a plain get() first')
        return self._get(self._next_url, params)

    def get_last(self, params=True):
        if self._last_url is None:
            raise AttributeError('Last url is not set, try querying a plain get() first')
        return self._get(self._last_url, params)
```

Is there any reason not to just change that?
